hpx/parallel/segmented_algorithms/find.hpp

```diff
@@ -0,0 +1,357 @@
+//  Copyright (c) 2017 Ajai V George
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_PARALLEL_SEGMENTED_ALGORITHM_FIND_JUN_22_2017_1157AM)
+#define HPX_PARALLEL_SEGMENTED_ALGORITHM_FIND_JUN_22_2017_1157AM
+
+#include <hpx/config.hpp>
+#include <hpx/traits/segmented_iterator_traits.hpp>
+
+#include <hpx/parallel/algorithms/detail/dispatch.hpp>
+#include <hpx/parallel/algorithms/find.hpp>
+#include <hpx/parallel/execution_policy.hpp>
+#include <hpx/parallel/segmented_algorithms/detail/dispatch.hpp>
+#include <hpx/parallel/util/detail/algorithm_result.hpp>
+#include <hpx/parallel/util/detail/handle_remote_exceptions.hpp>
+#include <hpx/parallel/util/projection_identity.hpp>
+
+#include <algorithm>
+#include <exception>
+#include <iterator>
+#include <list>
+#include <numeric>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace hpx { namespace parallel { inline namespace v1
+{
+    ///////////////////////////////////////////////////////////////////////////
+    // segmented_find
+    namespace detail
+    {
+        template <typename Algo, typename ExPolicy, typename InIter,
+            typename U>
+        inline typename std::enable_if<
```

Comment:
I'm not sure we need the `enable_if` here. This is an internal function in `detail` namespace and it is supposed to be called by an outer API function which performs the static verification already.